crates/storage/provider/src/traits/executor.rs

```diff
@@ -2,26 +2,27 @@
 
 use crate::{bundle_state::BundleStateWithReceipts, StateProvider};
 use reth_interfaces::executor::BlockExecutionError;
-use reth_primitives::{BlockNumber, BlockWithSenders, PruneModes, Receipt, U256};
+use reth_primitives::{BlockNumber, BlockWithSenders, ChainSpec, PruneModes, Receipt, U256};
 use std::time::Duration;
 use tracing::debug;
 
-/// Executor factory that would create the EVM with particular state provider.
-///
-/// It can be used to mock executor.
+/// A factory capable of creating an executor with the given state provider.
 pub trait ExecutorFactory: Send + Sync + 'static {
     /// Executor with [`StateProvider`]
     fn with_state<'a, SP: StateProvider + 'a>(
         &'a self,
-        _sp: SP,
+        sp: SP,
     ) -> Box<dyn PrunableBlockExecutor<Error = BlockExecutionError> + 'a>;
+
+    /// A handle to the chainspec used in this factory.
+    fn chain_spec(&self) -> &ChainSpec;
```

Comment:
needed for unwind

Reply:
this does not really belong here
we could also just give the exec stage the chainspec instead?

Reply:
removed